src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+                        // The value of $IsComposable is one of the Boolean literals true or false. Absence of the member means false.
+                        isComposable = propertyValue.ParseAsBoolean(context);
+                        break;
+
+                    default:
+                        // it MAY contain annotations.
+                        ParseCsdlAnnotation(propertyName, propertyValue, context, operationAnnotations);
+                        break;
+                }
+            });
+
+            CsdlLocation location = context.Location();
+            CsdlOperation csdlOperation;
+            if (isAction)
+            {
+                // It's action
+                csdlOperation = new CsdlAction(name, parameters, operationReturn, isBound == null ? false : isBound.Value, entitySetPath, location);
+            }
+            else
+            {
+                // It's function
+                csdlOperation = new CsdlFunction(name, parameters, operationReturn, isBound == null ? false : isBound.Value, entitySetPath,
+                    isComposable == null ? false : isComposable.Value, location);
+            }
+
+            operationAnnotations.ForEach(v => csdlOperation.AddAnnotation(v));
+            return csdlOperation;
+        }
+
+        internal static CsdlOperationParameter ParseCsdlParameter(JsonElement element, JsonParserContext context)
+        {
+            // A Parameter is Object
+            Debug.Assert(element.ValueKind == JsonValueKind.Object);
+            Debug.Assert(context != null);
+
+            CsdlTypeReference parameterType = CsdlJsonParseHelper.ParseCsdlTypeReference(element, context);
+
+            IList<CsdlAnnotation> prameterAnnotations = new List<CsdlAnnotation>();
+            string name = null;
+            element.ParseAsObject(context, (propertyName, propertyValue) =>
+            {
+                switch (propertyName)
+                {
+                    case "$Type":
+                    case "$Collection":
+                    case "$Nullable":
+                    case "$MaxLength":
+                    case "$Unicode":
+                    case "$Precision":
+                    case "$Scale":
+                    case "$SRID":
+                        // skip them because they are processed
+                        break;
+
+                    case "$Name":
+                        // The value of $Name is a string containing the parameter name.
+                        name = propertyValue.ParseAsString(context);
+                        break;
+
+                    default:
+                        ParseCsdlAnnotation(propertyName, propertyValue, context, prameterAnnotations);
+                        break;
+                }
+
+            });
+
+            // Workaround to process the optional parameter
+            bool isOptional = false;
+            string defaultValue = null;
+            // todo: Consider the namespace alias?

```

Comment:
>// todo: Consider the namespace alias? [](start = 12, length = 38)

yes; we should support namespace qualified.

Reply:
Create an issue to track on it : [https://github.com/OData/odata.net/issues/1818](https://github.com/OData/odata.net/issues/1818)

---
In reply to: [443697123](https://github.com/OData/odata.net/pull/1775#discussion_r443697123) [](ancestors = 443697123)